weedcoco/importers/deposit.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import pathlib
+import json
+import os
+import re
+from shutil import copyfile
+from functools import reduce
+from weedcoco.utils import get_image_average_hash
+from weedcoco.validation import ValidationError, validate
+
+
+def validate_duplicate_images(image_hash):
+    if len(set(image_hash.values())) != len(image_hash):
+        raise ValidationError("There are identical images in the image directory.")
+
+
+def validate_existing_images(repository_dir, image_hash):
+    addition_hash = set(image_hash.values())
+    all_existing_hash = get_all_existing_hash(repository_dir)
+    if len(all_existing_hash.union(addition_hash)) != len(all_existing_hash) + len(
+        addition_hash
+    ):
+        raise ValidationError("There are identical images in the repository.")
+
+
+def get_all_existing_hash(repository_dir):
+    return reduce(
+        lambda acc, it: acc.union(set(it[2])),
+        filter(lambda dir: dir[0].endswith("/images"), os.walk(repository_dir)),
+        set(),
+    )
+
+
+def setup_dataset_dir(repository_dir):
+    if not os.path.isdir(repository_dir):
+        os.mkdir(repository_dir)
+        dataset_dir = repository_dir / "dataset_0"
+    else:
+        if "dataset_0" in os.listdir(repository_dir):
+            latest_dataset_dir_index = max(
+                [
+                    int(dir.split("_")[-1])
+                    for dir in os.listdir(repository_dir)
+                    if re.fullmatch(r"^dataset_\d+$", dir)
+                ]
+            )
+            dataset_dir = repository_dir / f"dataset_{latest_dataset_dir_index + 1}"
+        else:
+            dataset_dir = repository_dir / "dataset_0"
+    os.mkdir(dataset_dir)
+    return dataset_dir
+
+
+def create_image_hash(image_dir):
+    return {
+        image_name: f'{get_image_average_hash(image_dir / image_name)}.{image_name.split(".")[-1]}'
+        for image_name in os.listdir(image_dir)
+        if image_name.lower().endswith(
+            (".png", ".jpg", ".jpeg", ".tiff", ".bmp", ".gif")
+        )
+    }
+
+
+def migrate_images(dataset_dir, raw_dir, image_hash):
+    image_dir = dataset_dir / "images"
+    if not os.path.isdir(image_dir):
+        os.mkdir(image_dir)
+
+    for image_origin in os.listdir(raw_dir):
+        image_path = dataset_dir / ("images/" + image_hash[image_origin])
+        if not os.path.isfile(image_path):
+            copyfile(raw_dir / image_origin, image_path)
+
+
+def generate_weedcoco(weedcoco_path, dataset_dir, image_hash):
+    with open(weedcoco_path) as f:
+        weedcoco = json.load(f)
+    validate(weedcoco)
```

Comment:
it might be appropriate to do the validation after the modification. We wouldn't want to inadvertently modify it in a way that broke the format.